backend-shared/config/db/util/application_info_cache_types.go

```diff
@@ -0,0 +1,46 @@
+package util
+
+import (
+	"context"
+	"time"
+
+	"github.com/redhat-appstudio/managed-gitops/backend-shared/config/db"
+)
+
+type ApplicationInfoCacheMessageType int
+
+type applicationStateCacheEntry struct {
+	appState        db.ApplicationState
+	cacheExpireTime time.Time // after this time, the entry should be removed from the cache.
```

Comment:
Looks like you are not currently setting `cacheExpireTime` in your PR (correct me if I am wrong). When the entry is added to the cache, `cacheExpireTime` should be set to `time.Now().Add(1 * time.Minute)`.

This allows the logic that you added elsewhere to work, because we can know when the cache was added using this value.